### PR TITLE
test: add test case for checking if all private methods are used.

### PR DESCRIPTION
**Pull request by MartinWitt**
Low hanging fruit related to #3631. Lets first finish the related issue and then review this. 
Next low hanging fruit after this is reduce the needed model builds to 1.

### CODE REVIEW

**Comment by monperrus** [comment]
Nice! It'd be great to start thinking of how we can share our generic 
architectural tests with the world.

**Comment by MartinWitt** [comment]
We could split our tests to a submodule like spoon-architecture and try to offer test case for often violated style problems.
A comparable product is PMD. There should be a lot of style violations we could check, but we would definitely need some good source for the problems. I'm unsure if there is some not outdated literature about it. But as it's near your research topic, I hope you can point to some literature for style problems. If there is any we can implement checks easily.

**Comment by monperrus** [comment]
For the record, https://github.com/INRIA/spoon/issues/3638 is for the new feature

**Comment by MartinWitt** [comment]
Architectural checks is the most exciting topic, because the other 3 are "solved" to some degree. Do you have some ideas for checks spoons is missing?

**Comment by MartinWitt** [comment]
Related to the issue I found the private method [getTypeName](https://github.com/INRIA/spoon/blob/c3881332d86069e06c8c31e370dae292affe4c1a/src/main/java/spoon/support/visitor/java/JavaReflectionTreeBuilder.java#L451) unused. I thought the methods has some serialization reasons, but can someone confirm the method is needed?

**Comment by monperrus** [comment]
> I thought the methods has some serialization reasons, but can someone confirm the method is needed?

The tests is the oracle. Just remove it and see whether CI is green.

**Comment by monperrus** [comment]
ping @cesarsotovalero @baudry this PR is pure debloat and @MartinWitt's test can be considered as a bloat detector.

**Comment by MartinWitt** [comment]
Looks like the CI is green, after removing the method. So lets keep it removed.

File: src/test/java/spoon/test/architecture/SpoonArchitectureEnforcerTest.java [hunk 1]
```diff
@@ -465,6 +472,39 @@ private static String computeDifference(Set<?> set1, Set<?> set2) {
...
 		return StringUtils.join(results, "\n");
 	}
+
+
+	private void checkPrivateMethodInvocations(CtModel model) {
+		List<CtMethod<?>> methods = model.getElements(new TypeFilter<>(CtMethod.class));
+		// only look at private methods
+		methods.removeIf(v -> !v.isPrivate());
+		// remove methods for serialization gods
+		methods.removeIf(v -> v.getSimpleName().matches("(readObject)|(readResolve)|(getTypeName)"));
```
Comment (monperrus) [inline src/test/java/spoon/test/architecture/SpoonArchitectureEnforcerTest.java:482]: `getTypeName` to be removed

**Comment by monperrus** [comment]
And one more cool architectural test in production! Thanks @MartinWitt

**Pull request merged**